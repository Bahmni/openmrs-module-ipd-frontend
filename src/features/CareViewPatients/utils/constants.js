export const searchKeys = ["bedNumber", "patientIdentifier", "patientName"];
export const items = [
    { id: "allTasks", text: "All Tasks" },
    { id: "medicationTasks", text: "Medication Tasks" },
    { id: "nonMedicationTasks", text: "Non-Medication Tasks" },
  ];
  