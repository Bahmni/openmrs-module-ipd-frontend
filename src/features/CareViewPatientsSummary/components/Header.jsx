import {
  epochTo12HourTimeFormat,
  epochTo24HourTimeFormat,
} from "../../../utils/DateTimeUtils";
import React, { useContext } from "react";
import propTypes from "prop-types";
import { CareViewContext } from "../../../context/CareViewContext";

export const Header = ({ timeframeLimitInHours, navHourEpoch }) => {
  const { ipdConfig } = useContext(CareViewContext);
  const { enable24HourTime = {} } = ipdConfig;
  return (
    <tr className="patient-row-container">
      <td
        className="patient-details-container"
        key="patient-details-header"
        data-testid="patient-details-header"
      ></td>
      {Array.from({ length: timeframeLimitInHours }, (_, i) => {
        const startTime = navHourEpoch.startHourEpoch + i * 3600;
        if (startTime < navHourEpoch.endHourEpoch) {
          let headerKey = `slot-details-header-${i}`;
          return (
            <td
              className="slot-details-header"
              key={headerKey}
              data-testid={headerKey}
            >
              <div className="time" data-testid={`time-frame-${i}`}>
                {enable24HourTime
                  ? epochTo24HourTimeFormat(startTime, true)
                  : epochTo12HourTimeFormat(startTime, true)}
              </div>
            </td>
          );
        }
      })}
    </tr>
  );
};

Header.propTypes = {
  timeframeLimitInHours: propTypes.number.isRequired,
  navHourEpoch: propTypes.object.isRequired,
};
