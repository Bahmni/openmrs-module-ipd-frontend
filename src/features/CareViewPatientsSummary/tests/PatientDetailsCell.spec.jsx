import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import { CareViewContext } from "../../../context/CareViewContext";
import { mockConfig } from "../../../utils/CommonUtils";
import "@testing-library/jest-dom/extend-expect";
import { PatientDetailsCell } from "../components/PatientDetailsCell";
import {
  mockParticipantData,
  mockPatientsList,
} from "./CareViewPatientsSummaryMock";

const mockHandleRefreshPatientList = jest.fn();

const mockContext = {
  careViewConfig: { timeframeLimitInHours: 2 },
  ipdConfig: mockConfig,
  provider: {
    uuid: "c1c26908-3f10-11e4-adec-0800271c1b75",
  },
  handleRefreshPatientList: mockHandleRefreshPatientList,
};

const mockBookmarkPatient = jest.fn();
const mockGetCurrentShiftTimes = jest.fn();

jest.mock("../../CareViewPatients/utils/CareViewPatientsUtils", () => ({
  bookmarkPatient: (payload) => mockBookmarkPatient(payload),
}));

jest.mock("../../../utils/DateTimeUtils", () => ({
  getCurrentShiftTimes: () => mockGetCurrentShiftTimes(),
}));

describe("PatientDetailsCell", () => {
  beforeEach(() => {
    jest.clearAllMocks();
    mockGetCurrentShiftTimes.mockReturnValue({
      startDateTime: 1710662400,
      endDateTime: 1710702000,
    });
    mockBookmarkPatient.mockResolvedValue(mockParticipantData);
  });

  it("should render patientDetailsCell component", () => {
    const { container } = render(
      <CareViewContext.Provider value={mockContext}>
        <PatientDetailsCell
          patientDetails={mockPatientsList.admittedPatients[0].patientDetails}
          bedDetails={mockPatientsList.admittedPatients[0].bedDetails}
          careTeamDetails={mockPatientsList.admittedPatients[0].careTeam}
          newTreatments={1}
          visitDetails={{ uuid: "sderf908-3f10-11e4-adec-0800271c1b72" }}
          navHourEpoch={{
            startHourEpoch: 1672575400,
            endHourEpoch: 1710511200,
          }}
        />
      </CareViewContext.Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders patient details correctly", async () => {
    const { queryByText } = render(
      <CareViewContext.Provider value={mockContext}>
        <PatientDetailsCell
          patientDetails={mockPatientsList.admittedPatients[0].patientDetails}
          bedDetails={mockPatientsList.admittedPatients[0].bedDetails}
          careTeamDetails={mockPatientsList.admittedPatients[0].careTeam}
          newTreatments={1}
          visitDetails={{ uuid: "sderf908-3f10-11e4-adec-0800271c1b72" }}
          navHourEpoch={{
            startHourEpoch: 1672575400,
            endHourEpoch: 1710511200,
          }}
        />
      </CareViewContext.Provider>
    );

    await waitFor(() => {
      expect(queryByText("PT51140")).toBeTruthy();
      expect(
        queryByText("AnnonFN-Jcilhyxuen AnnonMN-Dylkrgbpwo AnnonLN-Gkksnhzbeu")
      ).toBeTruthy();
      expect(queryByText("A-6")).toBeTruthy();
      expect(queryByText("14")).toBeTruthy();
    });
  });

  it("should show nurse name only if a nurse bookmarks a patient within the current shift", async () => {
    const container = render(
      <CareViewContext.Provider value={mockContext}>
        <PatientDetailsCell
          patientDetails={mockPatientsList.admittedPatients[0].patientDetails}
          bedDetails={mockPatientsList.admittedPatients[0].bedDetails}
          careTeamDetails={mockPatientsList.admittedPatients[0].careTeam}
          newTreatments={1}
          visitDetails={{ uuid: "sderf908-3f10-11e4-adec-0800271c1b72" }}
          navHourEpoch={{
            startHourEpoch: 1710509100,
            endHourEpoch: 1710511200,
          }}
        />
      </CareViewContext.Provider>
    );

    await waitFor(() => {
      expect(container).toMatchSnapshot();
      expect(container.queryByTestId("bookmark-filled-icon")).toBeTruthy();
      expect(container.queryByText("Nurse:")).toBeTruthy();
      expect(container.queryByText("Super Man")).toBeTruthy();
    });
  });

  it("should not show nurse name if the current time crossed the shift endTime", async () => {
    const container = render(
      <CareViewContext.Provider value={mockContext}>
        <PatientDetailsCell
          patientDetails={mockPatientsList.admittedPatients[0].patientDetails}
          bedDetails={mockPatientsList.admittedPatients[0].bedDetails}
          careTeamDetails={mockPatientsList.admittedPatients[0].careTeam}
          newTreatments={1}
          visitDetails={{ uuid: "sderf908-3f10-11e4-adec-0800271c1b72" }}
          navHourEpoch={{
            startHourEpoch: 1710509500,
            endHourEpoch: 1710609500,
          }}
        />
      </CareViewContext.Provider>
    );

    await waitFor(() => {
      expect(container).toMatchSnapshot();
      expect(container.queryByTestId("bookmark-add-icon")).toBeTruthy();
      expect(container.queryByText("Nurse:")).toBeFalsy();
      expect(container.queryByText("Super Man")).toBeFalsy();
    });
  });

  it("should call bookmark api with correct payload on click of bookmark add icon", async () => {
    const container = render(
      <CareViewContext.Provider value={mockContext}>
        <PatientDetailsCell
          patientDetails={mockPatientsList.admittedPatients[1].patientDetails}
          bedDetails={mockPatientsList.admittedPatients[1].bedDetails}
          careTeamDetails={mockPatientsList.admittedPatients[1].careTeam}
          newTreatments={1}
          visitDetails={{ uuid: "sderf908-3f10-11e4-adec-0800271c1b72" }}
          navHourEpoch={{
            startHourEpoch: 1710509100,
            endHourEpoch: 1710509100,
          }}
        />
      </CareViewContext.Provider>
    );

    fireEvent.click(container.queryByTestId("bookmark-add-icon"));

    await waitFor(() => {
      expect(container.queryByTestId("bookmark-filled-icon")).toBeTruthy();
      expect(mockBookmarkPatient).toHaveBeenCalledTimes(1);
      expect(mockBookmarkPatient).toHaveBeenCalledWith({
        careTeamParticipantsRequest: [
          {
            endTime: 1710702000,
            providerUuid: "c1c26908-3f10-11e4-adec-0800271c1b75",
            startTime: 1710662400,
          },
        ],
        patientUuid: "7278eb93-8c1d-4ef4-bcbf-4c6ee91365f7",
      });
    });
  });

  it("should call bookmark api with un bookmark payload on click of bookmark filled icon", async () => {
    const container = render(
      <CareViewContext.Provider value={mockContext}>
        <PatientDetailsCell
          patientDetails={mockPatientsList.admittedPatients[0].patientDetails}
          bedDetails={mockPatientsList.admittedPatients[0].bedDetails}
          careTeamDetails={mockPatientsList.admittedPatients[0].careTeam}
          newTreatments={1}
          visitDetails={{ uuid: "sderf908-3f10-11e4-adec-0800271c1b72" }}
          navHourEpoch={{
            startHourEpoch: 1710509100,
            endHourEpoch: 1710511200,
          }}
        />
      </CareViewContext.Provider>
    );

    fireEvent.click(container.queryByTestId("bookmark-filled-icon"));

    await waitFor(() => {
      expect(container.queryByTestId("bookmark-filled-icon")).not.toHaveClass(
        "bookmark-disabled"
      );
      expect(mockBookmarkPatient).toHaveBeenCalledTimes(1);
      expect(mockBookmarkPatient).toHaveBeenCalledWith({
        careTeamParticipantsRequest: [
          {
            uuid: "fc4d7a50-da94-4266-980e-eced24a64533",
            voided: true,
          },
        ],
        patientUuid: "17fd50c7-8f9e-48da-b9ed-88c1bd358798",
      });
    });
  });

  it("should disable bookmark filled icon when the patient is already bookmarked by another nurse", async () => {
    const container = render(
      <CareViewContext.Provider value={mockContext}>
        <PatientDetailsCell
          patientDetails={mockPatientsList.admittedPatients[2].patientDetails}
          bedDetails={mockPatientsList.admittedPatients[2].bedDetails}
          careTeamDetails={mockPatientsList.admittedPatients[2].careTeam}
          newTreatments={1}
          visitDetails={{ uuid: "sderf908-3f10-11e4-adec-0800271c1b72" }}
          navHourEpoch={{
            startHourEpoch: 1710509100,
            endHourEpoch: 1710511200,
          }}
        />
      </CareViewContext.Provider>
    );

    fireEvent.click(container.queryByTestId("bookmark-filled-icon"));

    await waitFor(() => {
      expect(container.queryByTestId("bookmark-filled-icon")).toBeTruthy();
      expect(container.queryByTestId("bookmark-filled-icon")).toHaveClass(
        "bookmark-disabled"
      );
    });
  });
});
