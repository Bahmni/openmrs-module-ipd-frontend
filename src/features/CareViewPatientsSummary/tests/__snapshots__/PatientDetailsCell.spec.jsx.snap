// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientDetailsCell should not show nurse name if the current time crossed the shift endTime 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <td
        class="patient-details-container"
      >
        <div
          class="care-view-patient-details"
        >
          <div
            class="admission-details"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M25,16H17a2.0023,2.0023,0,0,0-2,2v6H4V14H2V30H4V26H28v4h2V21A5.0059,5.0059,0,0,0,25,16Zm3,8H17V18h8a3.0033,3.0033,0,0,1,3,3Z"
              />
              <path
                d="M9.5 17A1.5 1.5 0 118 18.5 1.5017 1.5017 0 019.5 17m0-2A3.5 3.5 0 1013 18.5 3.5 3.5 0 009.5 15zM21 6L17 6 17 2 15 2 15 6 11 6 11 8 15 8 15 12 17 12 17 8 21 8 21 6z"
              />
            </svg>
            |
            <span>
              A-6
            </span>
            |
            <a
              class="bx--link"
              data-testid="identifier-ipd-dashboard"
              href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
            >
              <span>
                PT51140
              </span>
            </a>
          </div>
          <div>
            Patient
            :
             
            <span>
              AnnonFN-Jcilhyxuen AnnonMN-Dylkrgbpwo AnnonLN-Gkksnhzbeu
            </span>
             (
            <span>
              M
            </span>
            )
            <span
              class="separator"
            >
              |
            </span>
            <span>
              14
            </span>
            yrs
            <div
              class="treatments-notification"
            >
              <div
                class="warning_icon"
              >
                <svg
                  aria-hidden="true"
                  class="warning-icon-20"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 23a1.5 1.5 0 101.5 1.5A1.5 1.5 0 0016 23zM15 12H17V21H15z"
                  />
                  <path
                    d="M29,30H3a1,1,0,0,1-.8872-1.4614l13-25a1,1,0,0,1,1.7744,0l13,25A1,1,0,0,1,29,30ZM4.6507,28H27.3493l.002-.0033L16.002,6.1714h-.004L4.6487,27.9967Z"
                  />
                </svg>
                 
              </div>
              <div
                class="treatments-notification-span"
              >
                <div>
                  • 
                  1 New treatment(s): 
                  <a
                    class="bx--link"
                    data-testid="treatments-ipd-dashboard"
                    href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
                  >
                    Schedule Treatments
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          data-testid="bookmark-add-icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24,16V26.7515l-7.0962-3.5894L16,22.7051l-.9009.456L8,26.748V4H18V2H8A2,2,0,0,0,6,4V30l10-5.0537L26,30V16Z"
            />
            <path
              d="M26 6L26 2 24 2 24 6 20 6 20 8 24 8 24 12 26 12 26 8 30 8 30 6 26 6z"
            />
          </svg>
        </div>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="patient-details-container"
    >
      <div
        class="care-view-patient-details"
      >
        <div
          class="admission-details"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M25,16H17a2.0023,2.0023,0,0,0-2,2v6H4V14H2V30H4V26H28v4h2V21A5.0059,5.0059,0,0,0,25,16Zm3,8H17V18h8a3.0033,3.0033,0,0,1,3,3Z"
            />
            <path
              d="M9.5 17A1.5 1.5 0 118 18.5 1.5017 1.5017 0 019.5 17m0-2A3.5 3.5 0 1013 18.5 3.5 3.5 0 009.5 15zM21 6L17 6 17 2 15 2 15 6 11 6 11 8 15 8 15 12 17 12 17 8 21 8 21 6z"
            />
          </svg>
          |
          <span>
            A-6
          </span>
          |
          <a
            class="bx--link"
            data-testid="identifier-ipd-dashboard"
            href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
          >
            <span>
              PT51140
            </span>
          </a>
        </div>
        <div>
          Patient
          :
           
          <span>
            AnnonFN-Jcilhyxuen AnnonMN-Dylkrgbpwo AnnonLN-Gkksnhzbeu
          </span>
           (
          <span>
            M
          </span>
          )
          <span
            class="separator"
          >
            |
          </span>
          <span>
            14
          </span>
          yrs
          <div
            class="treatments-notification"
          >
            <div
              class="warning_icon"
            >
              <svg
                aria-hidden="true"
                class="warning-icon-20"
                fill="currentColor"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 23a1.5 1.5 0 101.5 1.5A1.5 1.5 0 0016 23zM15 12H17V21H15z"
                />
                <path
                  d="M29,30H3a1,1,0,0,1-.8872-1.4614l13-25a1,1,0,0,1,1.7744,0l13,25A1,1,0,0,1,29,30ZM4.6507,28H27.3493l.002-.0033L16.002,6.1714h-.004L4.6487,27.9967Z"
                />
              </svg>
               
            </div>
            <div
              class="treatments-notification-span"
            >
              <div>
                • 
                1 New treatment(s): 
                <a
                  class="bx--link"
                  data-testid="treatments-ipd-dashboard"
                  href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
                >
                  Schedule Treatments
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        data-testid="bookmark-add-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24,16V26.7515l-7.0962-3.5894L16,22.7051l-.9009.456L8,26.748V4H18V2H8A2,2,0,0,0,6,4V30l10-5.0537L26,30V16Z"
          />
          <path
            d="M26 6L26 2 24 2 24 6 20 6 20 8 24 8 24 12 26 12 26 8 30 8 30 6 26 6z"
          />
        </svg>
      </div>
    </td>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`PatientDetailsCell should render patientDetailsCell component 1`] = `
<div>
  <td
    class="patient-details-container"
  >
    <div
      class="care-view-patient-details"
    >
      <div
        class="admission-details"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="16"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M25,16H17a2.0023,2.0023,0,0,0-2,2v6H4V14H2V30H4V26H28v4h2V21A5.0059,5.0059,0,0,0,25,16Zm3,8H17V18h8a3.0033,3.0033,0,0,1,3,3Z"
          />
          <path
            d="M9.5 17A1.5 1.5 0 118 18.5 1.5017 1.5017 0 019.5 17m0-2A3.5 3.5 0 1013 18.5 3.5 3.5 0 009.5 15zM21 6L17 6 17 2 15 2 15 6 11 6 11 8 15 8 15 12 17 12 17 8 21 8 21 6z"
          />
        </svg>
        |
        <span>
          A-6
        </span>
        |
        <a
          class="bx--link"
          data-testid="identifier-ipd-dashboard"
          href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
        >
          <span>
            PT51140
          </span>
        </a>
      </div>
      <div>
        Patient
        :
         
        <span>
          AnnonFN-Jcilhyxuen AnnonMN-Dylkrgbpwo AnnonLN-Gkksnhzbeu
        </span>
         (
        <span>
          M
        </span>
        )
        <span
          class="separator"
        >
          |
        </span>
        <span>
          14
        </span>
        yrs
        <div
          class="treatments-notification"
        >
          <div
            class="warning_icon"
          >
            <svg
              aria-hidden="true"
              class="warning-icon-20"
              fill="currentColor"
              focusable="false"
              height="20"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 23a1.5 1.5 0 101.5 1.5A1.5 1.5 0 0016 23zM15 12H17V21H15z"
              />
              <path
                d="M29,30H3a1,1,0,0,1-.8872-1.4614l13-25a1,1,0,0,1,1.7744,0l13,25A1,1,0,0,1,29,30ZM4.6507,28H27.3493l.002-.0033L16.002,6.1714h-.004L4.6487,27.9967Z"
              />
            </svg>
             
          </div>
          <div
            class="treatments-notification-span"
          >
            <div>
              • 
              1 New treatment(s): 
              <a
                class="bx--link"
                data-testid="treatments-ipd-dashboard"
                href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
              >
                Schedule Treatments
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="bookmark-add-icon"
    >
      <svg
        aria-hidden="true"
        fill="currentColor"
        focusable="false"
        height="20"
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 32 32"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M24,16V26.7515l-7.0962-3.5894L16,22.7051l-.9009.456L8,26.748V4H18V2H8A2,2,0,0,0,6,4V30l10-5.0537L26,30V16Z"
        />
        <path
          d="M26 6L26 2 24 2 24 6 20 6 20 8 24 8 24 12 26 12 26 8 30 8 30 6 26 6z"
        />
      </svg>
    </div>
  </td>
</div>
`;

exports[`PatientDetailsCell should show nurse name only if a nurse bookmarks a patient within the current shift 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <td
        class="patient-details-container"
      >
        <div
          class="care-view-patient-details"
        >
          <div
            class="admission-details"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M25,16H17a2.0023,2.0023,0,0,0-2,2v6H4V14H2V30H4V26H28v4h2V21A5.0059,5.0059,0,0,0,25,16Zm3,8H17V18h8a3.0033,3.0033,0,0,1,3,3Z"
              />
              <path
                d="M9.5 17A1.5 1.5 0 118 18.5 1.5017 1.5017 0 019.5 17m0-2A3.5 3.5 0 1013 18.5 3.5 3.5 0 009.5 15zM21 6L17 6 17 2 15 2 15 6 11 6 11 8 15 8 15 12 17 12 17 8 21 8 21 6z"
              />
            </svg>
            |
            <span>
              A-6
            </span>
            |
            <a
              class="bx--link"
              data-testid="identifier-ipd-dashboard"
              href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
            >
              <span>
                PT51140
              </span>
            </a>
          </div>
          <div>
            Patient
            :
             
            <span>
              AnnonFN-Jcilhyxuen AnnonMN-Dylkrgbpwo AnnonLN-Gkksnhzbeu
            </span>
             (
            <span>
              M
            </span>
            )
            <span
              class="separator"
            >
              |
            </span>
            <span>
              14
            </span>
            yrs
            <div
              class="treatments-notification"
            >
              <div
                class="warning_icon"
              >
                <svg
                  aria-hidden="true"
                  class="warning-icon-20"
                  fill="currentColor"
                  focusable="false"
                  height="20"
                  preserveAspectRatio="xMidYMid meet"
                  viewBox="0 0 32 32"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M16 23a1.5 1.5 0 101.5 1.5A1.5 1.5 0 0016 23zM15 12H17V21H15z"
                  />
                  <path
                    d="M29,30H3a1,1,0,0,1-.8872-1.4614l13-25a1,1,0,0,1,1.7744,0l13,25A1,1,0,0,1,29,30ZM4.6507,28H27.3493l.002-.0033L16.002,6.1714h-.004L4.6487,27.9967Z"
                  />
                </svg>
                 
              </div>
              <div
                class="treatments-notification-span"
              >
                <div>
                  • 
                  1 New treatment(s): 
                  <a
                    class="bx--link"
                    data-testid="treatments-ipd-dashboard"
                    href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
                  >
                    Schedule Treatments
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div>
            Nurse
            :
             
            <span>
              Super Man
            </span>
          </div>
        </div>
        <div
          data-testid="bookmark-filled-icon"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="20"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M24,2H8A2,2,0,0,0,6,4V30l10-5.0538L26,30V4A2,2,0,0,0,24,2Z"
            />
          </svg>
        </div>
      </td>
    </div>
  </body>,
  "container": <div>
    <td
      class="patient-details-container"
    >
      <div
        class="care-view-patient-details"
      >
        <div
          class="admission-details"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="16"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 32 32"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M25,16H17a2.0023,2.0023,0,0,0-2,2v6H4V14H2V30H4V26H28v4h2V21A5.0059,5.0059,0,0,0,25,16Zm3,8H17V18h8a3.0033,3.0033,0,0,1,3,3Z"
            />
            <path
              d="M9.5 17A1.5 1.5 0 118 18.5 1.5017 1.5017 0 019.5 17m0-2A3.5 3.5 0 1013 18.5 3.5 3.5 0 009.5 15zM21 6L17 6 17 2 15 2 15 6 11 6 11 8 15 8 15 12 17 12 17 8 21 8 21 6z"
            />
          </svg>
          |
          <span>
            A-6
          </span>
          |
          <a
            class="bx--link"
            data-testid="identifier-ipd-dashboard"
            href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
          >
            <span>
              PT51140
            </span>
          </a>
        </div>
        <div>
          Patient
          :
           
          <span>
            AnnonFN-Jcilhyxuen AnnonMN-Dylkrgbpwo AnnonLN-Gkksnhzbeu
          </span>
           (
          <span>
            M
          </span>
          )
          <span
            class="separator"
          >
            |
          </span>
          <span>
            14
          </span>
          yrs
          <div
            class="treatments-notification"
          >
            <div
              class="warning_icon"
            >
              <svg
                aria-hidden="true"
                class="warning-icon-20"
                fill="currentColor"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 23a1.5 1.5 0 101.5 1.5A1.5 1.5 0 0016 23zM15 12H17V21H15z"
                />
                <path
                  d="M29,30H3a1,1,0,0,1-.8872-1.4614l13-25a1,1,0,0,1,1.7744,0l13,25A1,1,0,0,1,29,30ZM4.6507,28H27.3493l.002-.0033L16.002,6.1714h-.004L4.6487,27.9967Z"
                />
              </svg>
               
            </div>
            <div
              class="treatments-notification-span"
            >
              <div>
                • 
                1 New treatment(s): 
                <a
                  class="bx--link"
                  data-testid="treatments-ipd-dashboard"
                  href="/bahmni/clinical/#/default/patient/17fd50c7-8f9e-48da-b9ed-88c1bd358798/dashboard/visit/ipd/sderf908-3f10-11e4-adec-0800271c1b72?source=careViewDashboard"
                >
                  Schedule Treatments
                </a>
              </div>
            </div>
          </div>
        </div>
        <div>
          Nurse
          :
           
          <span>
            Super Man
          </span>
        </div>
      </div>
      <div
        data-testid="bookmark-filled-icon"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          focusable="false"
          height="20"
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 32 32"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M24,2H8A2,2,0,0,0,6,4V30l10-5.0538L26,30V4A2,2,0,0,0,24,2Z"
          />
        </svg>
      </div>
    </td>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
